media-gfx/oxipng: add 9.1.3

lo48576 opened the pull request:
<!-- Please put the pull request description above -->

Add media-gfx/oxipng-9.1.3.

Additionally, `system-libdeflate` USE flag is added (and is enabled by default) to let users choose whether to use app-arch/libdeflate or the bundled one.
I haven't done this at oxipng-9.1.2-r2 and have forced system libdeflate because of [bug 944285](https://bugs.gentoo.org/944285) on arm64, but oxipng-9.1.3 shouldn't suffer with it (<https://github.com/shssoichiro/oxipng/pull/659#pullrequestreview-2476559704>), so I believe it is good timing to add the USE flag.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

thesamesam commented on media-gfx/oxipng/files/oxipng-9.1.3-add-system-libdeflate-feature.patch:
```diff
@@ -0,0 +1,47 @@
+From 161b184373092742e9867a8b8cabad92a0395f15 Mon Sep 17 00:00:00 2001
```
Please add a link to the upstream source/reference for this patch to the top of the file.

lo48576 replied:
Done!
I didn't know that the leading block can contain (almost) free-form text...

thesamesam commented on media-gfx/oxipng/oxipng-9.1.3.ebuild:
```diff
@@ -0,0 +1,157 @@
+PATCHES=( "${FILESDIR}/oxipng-9.1.3-add-system-libdeflate-feature.patch" )
+IUSE="+system-libdeflate"
+
+BDEPEND="system-libdeflate? ( virtual/pkgconfig )"
```
Is there a reason to not do this unconditionally?

juippis replied:
Yeah let's use the system lib unconditionally until there's a reason not to. 

Firefox only offers a choice out of necessity, or because Mozilla also patches their bundled libs to provide better performance. But not every lib is given to `system*` use flag, some system deps are unconditionally forced.

juippis commented on media-gfx/oxipng/oxipng-9.1.3.ebuild:
```diff
@@ -0,0 +1,157 @@
+		# Also, don't forget updating `DEPNED` and `RDEPEND` in sync, or libdeflater
+		# crate will fail to pick system library and silently use the bundled
+		# version of libdeflate.
+		( cd "${WORKDIR}"/cargo_home/gentoo/libdeflate-sys-1.22.0 && \
+			eapply "${FILESDIR}"/libdeflater-1.22.0-relax-libdeflate-sys-version.patch )
```
Let's open this up like we like to, and have proper error-handling here:
```suggestion
        cd "${WORKDIR}"/cargo_home/gentoo/libdeflate-sys-1.22.0 || die
        eapply "${FILESDIR}"/libdeflater-1.22.0-relax-libdeflate-sys-version.patch
```
although maybe you can edit the patch file itself to handle this path automatically.

lo48576 replied:
> although maybe you can edit the patch file itself to handle this path automatically.

I was able to put `/cargo_home/gentoo/libdeflate-sys-1.22.0` into the patch file, but wasn't possible to put it to `$PATCHES`, since the patch target file was in `${WORKDIR}/cargo_home/...` but the working directory was `${S}`, which is `${WORKDIR}/${P}` by default.

lo48576 commented:
* Rebased onto the current master to use dev-lang/rust-1.84.0 on building.
* Made system libdeflate dependency unconditional.
* Updated a patch to contain some part of the target file path, and put a comment into it instead of inside ebuild.
    + Followed the recommendation from [Patches – Gentoo Development Guide](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html).

juippis commented on media-gfx/oxipng/oxipng-9.1.3.ebuild:
```diff
@@ -0,0 +1,144 @@
+QA_PRESTRIPPED="usr/bin/${PN}"
+
+src_prepare() {
+	cd "${WORKDIR}"
+	eapply "${FILESDIR}"/oxipng-9.1.3-libdeflater-1.22.0-relax-libdeflate-sys-version.patch
+	cd -
```
Well, you could've avoided this if you specified correct path directly in the patch file - but I see this was by choice. I'll add some error-handling and this is good!

juippis reviewed (approved):
Sorry for the delay and thanks!
